demo/dotnet/PvSpeakerDemo/PvSpeakerDemo.cs

```diff
@@ -0,0 +1,246 @@
+
+            bool showAudioDevices = false;
+            int audioDeviceIndex = -1;
+            string inputWavPath = null;
+            int bufferSizeSecs = 20;
+            string outputWavPath = null;
+
+            int argIndex = 0;
+            while (argIndex < args.Length)
+            {
+                if (args[argIndex] == "--show_audio_devices")
+                {
+                    showAudioDevices = true;
+                    argIndex++;
+                }
+                else if (args[argIndex] == "--audio_device_index")
+                {
+                    if (++argIndex < args.Length && int.TryParse(args[argIndex], out int deviceIndex))
+                    {
+                        audioDeviceIndex = deviceIndex;
+                        argIndex++;
+                    }
+                }
+                else if (args[argIndex] == "--input_wav_path")
+                {
+                    if (++argIndex < args.Length)
+                    {
+                        inputWavPath = args[argIndex++];
+                    }
+                }
+                else if (args[argIndex] == "--buffer_size_secs")
+                {
+                    if (++argIndex < args.Length && int.TryParse(args[argIndex], out int bufferSize))
+                    {
+                        bufferSizeSecs = bufferSize;
+                        argIndex++;
+                    }
+                }
+                else if (args[argIndex] == "--output_wav_path")
+                {
+                    if (++argIndex < args.Length)
+                    {
+                        outputWavPath = args[argIndex++];
+                    }
+                }
+                else
+                {
+                    argIndex++;
+                }
+            }
+
+            if (showAudioDevices)
+            {
+                ShowAudioDevices();
+                return;
+            }
+
+            try
+            {
+                WavFileInfo wavInfo = GetWavFileInfo(inputWavPath);
+
+                using (PvSpeaker speaker = PvSpeaker.Create(
+                        sampleRate: wavInfo.SampleRate,
+                        bitsPerSample: wavInfo.BitsPerSample,
+                        bufferSizeSecs: bufferSizeSecs,
+                        deviceIndex: audioDeviceIndex))
+                {
+                    Console.CancelKeyPress += delegate (object sender, ConsoleCancelEventArgs e)
+                    {
+                        e.Cancel = true;
+                        speaker.Stop();
+                        Console.WriteLine("Stopping...");
+                    };
+
+                    speaker.Start();
+
+                    if (outputWavPath != null)
+                    {
+                        speaker.WriteToFile(outputWavPath);
+                    }
+
+                    Console.WriteLine($"Using PvSpeaker version: {speaker.Version}");
+
+                    int bytesPerSample = wavInfo.BitsPerSample / 8;
```

Comment:
might also be nice to say `Playing ${input_audio_filename}...` so that the user knows something should be happening